etcd/cmd/microshift-etcd/run.go

```diff
@@ -0,0 +1,117 @@
+package main
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"syscall"
+
+	"github.com/openshift/microshift/pkg/config"
+	"github.com/openshift/microshift/pkg/util/cryptomaterial"
+
+	"github.com/spf13/cobra"
+	etcd "go.etcd.io/etcd/server/v3/embed"
+	"k8s.io/klog/v2"
+)
+
+func NewRunEtcdCommand() *cobra.Command {
+	cfg := config.NewMicroshiftConfig()
+	cmd := &cobra.Command{
+		Use: "run",
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			if err := cfg.ReadAndValidate(config.GetConfigFile(), cmd.Flags()); err != nil {
+				klog.Fatalf("Error in reading and validating flags", err)
+			}
+
+			e := NewEtcd(cfg)
+			return e.Run()
+		},
+	}
+	return cmd
+}
+
+var tlsCipherSuites = []string{
+	"TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384",
+	"TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384",
+	"TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256",
+	"TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
+	"TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305",
+	"TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305",
+}
+
+type EtcdService struct {
+	etcdCfg *etcd.Config
+}
+
+func NewEtcd(cfg *config.MicroshiftConfig) *EtcdService {
+	s := &EtcdService{}
+	s.configure(cfg)
+	return s
+}
+
+func (s *EtcdService) Name() string { return "etcd" }
+
+func (s *EtcdService) configure(cfg *config.MicroshiftConfig) {
+	microshiftDataDir := config.GetDataDir()
+	certsDir := cryptomaterial.CertsDirectory(microshiftDataDir)
+
+	etcdServingCertDir := cryptomaterial.EtcdServingCertDir(certsDir)
+	etcdPeerCertDir := cryptomaterial.EtcdPeerCertDir(certsDir)
+	etcdSignerCertPath := cryptomaterial.CACertPath(cryptomaterial.EtcdSignerDir(certsDir))
+	dataDir := filepath.Join(microshiftDataDir, s.Name())
+
+	// based on https://github.com/openshift/cluster-etcd-operator/blob/master/bindata/bootkube/bootstrap-manifests/etcd-member-pod.yaml#L19
+	s.etcdCfg = etcd.NewConfig()
+	s.etcdCfg.ClusterState = "new"
+	//s.etcdCfg.ForceNewCluster = true //TODO
+	s.etcdCfg.Logger = "zap"
+	s.etcdCfg.Dir = dataDir
+	s.etcdCfg.APUrls = setURL([]string{cfg.NodeIP}, ":2380")
```

Comment:
Does this need to be the public node IP? Could it be localhost, for example, like the other endpoints?

Reply:
Looks like `127.0.0.1` works